packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart

```diff
@@ -372,3 +396,44 @@ class VideoPlayerOptions {
   /// currently no way to implement this feature in this platform).
   final bool mixWithOthers;
 }
+
+/// Subtitle option which embedded into video.
+class EmbeddedSubtitle {
+  /// Subtitle option which embedded into video.
+  ///
+  /// * It's recommended to not create direct instance by this constructor. Try to
+  ///   get from video player controller.
+  const EmbeddedSubtitle({
+    required this.language,
+    required this.label,
+    required this.trackIndex,
+    required this.groupIndex,
+    required this.renderIndex,
+  });
+
+  /// An instance of embedded subtitle which use to disable subtitle stream.
+  const EmbeddedSubtitle.none()
+      : language = null,
+        label = null,
+        trackIndex = null,
+        groupIndex = null,
+        renderIndex = null;
+
+  /// Language of embedded subtitle
+  final String? language;
+
+  /// Label associated for each option by embedded subtitle
+  final String? label;
+
+  /// Subtitle track index between list of subtitle in embedded subtitle
+  final int? trackIndex;
+
+  /// Subtitle group track index between available track for video.
+  final int? groupIndex;
+
+  /// Subtitle group track render index
+  final int? renderIndex;
+
+  /// Checks either embedded subtitle is selected or is for remove subtitle
```

Comment:
Checks whether the embedded subtitle is selected or is for removing the subtitle.